### PR TITLE
Enhancement: Implement Alert Modal Integration in Loading Screen

### LINKED ISSUE #201: Warn user when browser features are missing
In firefox the keyboard layout detection is not supported. Using the new modal system to warn the user that QWERTY will be assumed is necessary.

### PR DESCRIPTION
**Pull request by nakul-py**
closes #201

### CODE REVIEW

**Review by BarthPaleologue — CHANGES_REQUESTED** [review]
Looking good! Just a small nitpicking before merging, I would like the modal to be awaited here instead: 
https://github.com/BarthPaleologue/CosmosJourneyer/blob/main/src/ts/cosmosJourneyer.ts#L319

(That's my bad, I was not precise enough when describing the issue)

This will allow to show the modal only once instead of after each loading screen :)

**Comment by nakul-py** [comment]
- This commit will close this issue https://github.com/BarthPaleologue/CosmosJourneyer/pull/207/commits/8e2bf9273edad64f911cdbbd9c265a63f9cb774f

- closes #201

File: src/ts/cosmosJourneyer.ts [hunk 1]
```diff
@@ -314,6 +314,8 @@ export class CosmosJourneyer {
         await starSystemView.initAssets();
         starSystemView.resetPlayer();
 
+        await alertModal("Loading Complete! Welcome to Cosmos Journeyer.");
```
Comment (BarthPaleologue) [inline src/ts/cosmosJourneyer.ts:317]: Yup that's the right place! But you still need to check for `navigator.keyboard` in order to only display the modal with the message about the QWERTY layout being assumed by default when it is not supported.

**Review by BarthPaleologue — CHANGES_REQUESTED** [review]

**Comment by nakul-py** [comment]
This commit will  close issue https://github.com/BarthPaleologue/CosmosJourneyer/pull/207/commits/f6d46b0705808b9d77080a12c9dcd46f1a4810e5

File: src/ts/cosmosJourneyer.ts [hunk 2]
```diff
@@ -314,6 +314,14 @@ export class CosmosJourneyer {
         await starSystemView.initAssets();
         starSystemView.resetPlayer();
 
+        if (!navigator.keyboard) {
+            await alertModal(
+                "Your keyboard layout could not be detected. The QWERTY layout will be assumed by default."
+            );
+        }
+
+        await alertModal("Loading Complete! Welcome to Cosmos Journeyer.");
```
Comment (BarthPaleologue) [inline src/ts/cosmosJourneyer.ts:323]: You forgot to remove this loading complete modal. The loading complete message is already communicated to the player by the progress bar.

**Review by BarthPaleologue — CHANGES_REQUESTED** [review]

**Comment by nakul-py** [comment]
Now all ok https://github.com/BarthPaleologue/CosmosJourneyer/pull/207/commits/f5f19fe3e0a0888494bfc84185aad5bac36e9b9b

**Review by BarthPaleologue — APPROVED** [review]

**Comment by BarthPaleologue** [comment]
LGTM!

**Comment by BarthPaleologue** [comment]
To format your code, use `npm run format` or `pnpm format` and then we can merge it 👍

**Comment by BarthPaleologue** [comment]
I think you should do `git pull` before formatting because I fixed a merged conflict between your branch and main. By force-pushing to your branch, you removed my conflict resolution

**Comment by BarthPaleologue** [comment]
Most code editors provide a format on save option now, that could save you a lot of time and trouble in the future! 

Thanks for the contrib anyway, I appreciate it a lot :)

**Pull request merged**